internal/server/scim.go

```diff
@@ -22,6 +25,13 @@ func ListProviderUsers(c *gin.Context, r *api.SCIMParametersRequest) (*api.ListP
 		StartIndex: r.StartIndex,
 		Count:      r.Count,
 	}
+	if r.Filter != "" {
+		exp, err := filter.ParseFilter([]byte(r.Filter))
+		if err != nil {
+			return nil, fmt.Errorf("parse SCIM filter expression: %w", err)
+		}
+		p.Filter = &exp
```

Comment:
```suggestion
		p.Filter = exp
```

If we make this not a pointer we no longer need to take the address